roles/assemble_template/tasks/main.yml

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Cloudera, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: Discover fragment files
+  ansible.builtin.find:
+    paths: "{{ cluster_template_fragments_directory | mandatory  }}"
+    patterns: "{{ json_regex_pattren | default(omit) }}"
+    use_regex: yes
+    recurse: yes
+  register: fragments
+
+- name: Create a temporary directory
+  ansible.builtin.tempfile:
```

Comment:
This temp directory should get cleaned up by the role. See below comment for an idea.

Reply:
Added